src/Service/VendorService/EbookCentral/EbookCentralVendorService.php

```diff
@@ -75,15 +76,16 @@ public function load(): VendorImportResultMessage
 
             foreach ($reader->getSheetIterator() as $sheet) {
                 foreach ($sheet->getRowIterator() as $row) {
+                    $cellsArray = $row->getCells();
                     if (0 === $totalRows) {
-                        if ('PrintIsbn' !== $row[2] || 'EIsbn' !== $row[3] || 'http://ebookcentral.proquest.com/covers/Document ID-l.jpg' !== $row[6]) {
+                        if ('PrintIsbn' !== $cellsArray[2]->getValue() || 'EIsbn' !== $cellsArray[3]->getVAlue() || 'http://ebookcentral.proquest.com/covers/Document ID-l.jpg' !== $cellsArray[6]->getValue()) {
```

Comment:
does $cellsArray[2], $cellsArray[3] and $cellsArray[6] always exist?

Question applies to all other places in the code

Reply:
Yes/No. We are importing a spreadsheet. Spreadsheet format can change over time. The purpose of this check is to get an error if the relevant columns are not what we expect.

Reply:
Ok... so we accept exceptions due to nonexistent array entries. This will pose a possible issue if run automatically, but issues will be caught if run manually

Reply:
When executed via cron-jobs the last 3 failed and successful runs are saved. So we can see the logs. @tuj